builder/etcd/etcd.go

```diff
@@ -0,0 +1,529 @@
+// 	- retries (int): Number of times to retry a connection to the server
+// 	- retrySleep (time.Duration): How long to sleep between retries
+//
+// Returns:
+// 	This puts an *etcd.Client into the context.
+func CreateClient(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+	url := p.Get("url", "http://localhost:4001").(string)
+
+	// Backed this out because it's unnecessary so far.
+	//hosts := p.Get("urls", []string{"http://localhost:4001"}).([]string)
+	hosts := []string{url}
+	retryCycles = p.Get("retries", retryCycles).(int)
+	retrySleep = p.Get("retrySleep", retrySleep).(time.Duration)
+
+	// Support `host:port` format, too.
+	for i, host := range hosts {
+		if !strings.Contains(host, "://") {
+			hosts[i] = "http://" + host
+		}
+	}
+
+	client := etcd.NewClient(hosts)
+	client.CheckRetry = checkRetry
+
+	return client, nil
+}
+
+// Get performs an etcd Get operation.
+//
+// Params:
+// 	- client (EtcdGetter): Etcd client
+// 	- path (string): The path/key to fetch
+//
+// Returns:
+// - This puts an `etcd.Response` into the context, and returns an error
+//   if the client could not connect.
+func Get(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+	cli, ok := p.Has("client")
+	if !ok {
+		return nil, errors.New("No Etcd client found.")
+	}
+	client := cli.(Getter)
+	path := p.Get("path", "/").(string)
+
+	res, err := client.Get(path, false, false)
+	if err != nil {
+		return res, err
+	}
+
+	if !res.Node.Dir {
+		return res, fmt.Errorf("Expected / to be a dir.")
+	}
+	return res, err
+}
+
+// IsRunning checks to see if etcd is running.
+//
+// It will test `count` times before giving up.
+//
+// Params:
+// 	- client (EtcdGetter)
+// 	- count (int): Number of times to try before giving up.
+//
+// Returns:
+// 	boolean true if etcd is listening.
+func IsRunning(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+	client := p.Get("client", nil).(Getter)
+	count := p.Get("count", 20).(int)
+	for i := 0; i < count; i++ {
+		_, err := client.Get("/", false, false)
+		if err == nil {
+			return true, nil
+		}
+		log.Infof(c, "Waiting for etcd to come online.")
+		time.Sleep(250 * time.Millisecond)
+	}
+	log.Errf(c, "Etcd is not answering after %d attempts.", count)
+	return false, &cookoo.FatalError{"Could not connect to Etcd."}
+}
+
+// Set sets a value in etcd.
+//
+// Params:
+// 	- key (string): The key
+// 	- value (string): The value
+// 	- ttl (uint64): Time to live
+// 	- client (EtcdGetter): Client, usually an *etcd.Client.
+//
+// Returns:
+// 	- *etcd.Result
+func Set(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+	key := p.Get("key", "").(string)
+	value := p.Get("value", "").(string)
+	ttl := p.Get("ttl", uint64(20)).(uint64)
+	client := p.Get("client", nil).(Setter)
+
+	res, err := client.Set(key, value, ttl)
+	if err != nil {
+		log.Infof(c, "Failed to set %s=%s", key, value)
+	}
+
+	return res, err
```

Comment:
This one is actually not an error. I return an error on purpose, because it may not be `nil`. But I guess I could change line 161 to just return the error there. That's probably easier to read.